### ISSUE TITLE
Timecode in Transcript?

### ISSUE THREAD

**Issue by Paddywack0** [octimot/StoryToolkitAI#20]
Managed to get this working today and it is utterly brilliant! Thankyou.
Would it be possible to add Timeline or Source TC to the Transcribed file so that when I pass it to an Edit Producer or Director etc. They can pull sections on a script with the TC. Then I can go to the edit and easily find sections by TC.

This is how I have done this before on Trint etc. and works pretty well. Unless you have a better way!

thanks again
Nick

**Comment by octimot** [comment]
Hey Nick!

That's a good idea! I'll try to make include it into an update as soon as possible. 

We're so used to working inside the tool that we didn't think that's needed...

BTW, if you're dealing with producers who like to keep things digital, the app should work without Resolve. So, you can pretty much send the transcript.json file to anyone who has the app on their machine and they can edit, and most importantly, search semantically for content (via advanced search).

Cheers!

**Comment by Paddywack0** [comment]
That would be excellent! If there was a way that a Producer could link and sync the JSON to the exported Video file (assuming Resolve has exported a Video file) that would be amazing especially with remote working. I could send them the JSON and the video file and they could watch and annotate as they go!

**Comment by Paddywack0** [comment]
Yes. So that Producers can view without a copy of Resolve. Either a player in the tool or a link to something like VLC?

**Comment by octimot** [comment]
Yea! VLC could work faster as an implementation. I'll look for options.

**Comment by Paddywack0** [comment]
Having a built in player is how programmes like Speedscriber or Trint work. Obviously as an editor I am chuffed with the way it works within Resolve. I paid out a fortune for the original Phrasefind in Avid only for it to become redundant when they went 64bit (you had to pay for the new version.)

Most of the producers I work with are happy to use VLC and it is cross platform.

**Comment by Paddywack0** [comment]
BTW - is this the right place to make suggestions like this? They are feature requests rather than Issues!
I have lots more but want to give the tool a good run tomorrow (I'm in the UK and it's late!).

Really amazing work! BM should pay you to implement it inside Resolve.

**Comment by Paddywack0** [comment]
Back to the Timecode request I notice having a play this morning that Shift+C  gives you the ability to cut and paste a selection with a Timecode so I'm assuming it wouldn't be too difficult to cut and paste or Export an entire transcript with Timecodes throughout. (This coming from someone who knows nothing of coding!)

**Comment by octimot** [comment]
> Most of the producers I work with are happy to use VLC and it is cross platform.

That's good to know. I'll see if this is possible and plan an update soon. We're also thinking about potentially synching transcripts between team members in the cloud where you could also have a basic video player - but that would be a major investment, so we need to see how it could work...

>  I notice having a play this morning that Shift+C gives you the ability to cut and paste a selection with a Timecode so I'm assuming it wouldn't be too difficult to cut and paste or Export an entire transcript with Timecodes

A basic implementation is quite trivial since it's already working for the subtitles (SRT) and SHIFT+C copy function. BTW, if you have line gaps between the selected blocks of text, each block should be preceded by its own timecode when pasting outside of the tool. 

The only question is how often should the Timecodes appear next to the text when you're exporting the whole transcript? 

Technically, we could show a timecode per line of transcript (like we do for subtitles, or perhaps per phrase, considering phrases are divided by punctuation marks). We could even have two columns (timecodes on the left and text on the right), for fast pasting into a spreadsheet or similar.

On the far end of the spectrum we could show timecodes every few seconds (let's say cca. every 5 seconds) of text - this would also be fast to implement.

But ideally, we would probably show them based on ideas / speakers etc. and this requires some more features to be developed in the app (hopefully in the very near future).

> Also...is there a way of changing the default Font size and maybe the font in the transcript viewer. It is a bit on the large size for me on my screen setup.

Not in the current standalone release, but most likely in an update this or next week.

**Comment by Paddywack0** [comment]
For me having TC at the start of each Section (phrase?). i.e. the section that gets highlighted when you click on it.

For most workflows I have seen a fairly basic implementation would be fine. The following is from a Trint script:

> Speaker 2 [00:01:14]
> Hi. Pleased to meet you.
> 
> Speaker 1 [00:01:16]
> Likewise. No bad place you got here?
> 
> Speaker 2 [00:01:18]
> No, it's lovely. Yeah, we're very lucky. Look.
> 
> Speaker 1 [00:01:22]
> Badges. So when did your love affair with badges begin?

This just uses a frame count rather than actual Timeline TC. In the original the TC and Speaker Name are highlighted in a different colour.

My workflow would be to give the Timecoded Transcript to my Producer (preferably with a synced Video) - they would then either highlight sections of the transcript or copy and paste sections into a script. I would then take that script and go to the Timecodes and find the cuts they wanted. StoryAI would make that much simpler in that I can see the transcript synced and/or search for sections.

**Comment by octimot** [comment]
In your example, the text seems to be split by speaker, since the phrases are grouped together - this is how most providers format the transcripts (simonsays, otter.ai etc.). In our case, since we don't have the speaker recognition available yet, this will look a bit messy and potentially make the transcript hard to follow by the reader.

But, in the next update, CMD+SHIFT+C will copy the lines with their timecodes to the clipboard (or seconds, if Resolve is not turned on). They will look like this:

```
01:00:10:00	I'd like you to meet Dr. Kalinin.
01:00:12:00	How do you do?
01:00:14:00	Dr. Sretniver.
01:00:16:00	And this is Dr. Andrei Smyslov.
01:00:18:00	Oh, how do you do?
01:00:20:00	I've heard a lot about you.
01:00:22:00	Would you sit down?
```

By using the tab space between the timecode and the line, these could be either used as plain text, or even pasted into a spreadsheet.

As soon as we add transcript segments grouping and speaker recognition, we can come back to this and see if it's worth upgrading to the format you mentioned, alright?

**Comment by octimot** [comment]
> TBH - this would work for me especially if it is TAB delimited so you can put it into columns/tables whatever.

Great!

> I will prob wait for the standalone Win10 release as I'm not confident installing the code. 

It's coming this or next week. We have to finish and test some things in our editing room before making the release bundles.

> Presumably the new release will be able to pull timecode out of transcripts done with the previous release?

Of course, no work you've done will be in vain from this point of view.

**Comment by Paddywack0** [comment]
I'm on 0.19.5 What is the difference between Copy with TC and Copy Block with TC?
If I paste the first into a Google Doc I get all the text with TC on each line/sentence but the second just seems to paste the Transcript with no TC except one at the very beginning. Is this correct?

**Comment by octimot** [comment]
@Paddywack0 

If you Copy with Block TC, you'll get a timecode on top of each block of lines which go together uninterrupted.

Try to add a gap between the lines that you select (unselect some lines in the middle of the selection) and Copy with Block TC then and paste it somewhere to see what I mean.

Cheers!

**Comment by Paddywack0** [comment]
OK. I got it. Great for copying chunks into a script.
is there any way that the normal Copy with TC could have a value for how many times the TC appears. At the moment it is the beginning of every line which can be a bit busy. It would be great if you could choose every 3rd line or 5th line etc.

**Comment by octimot** [comment]
This is in our todo backlog already, but we'll try to make it work soon!

Cheers
